kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualCluster.java

```
@@ -229,6 +259,45 @@ private static NettyTrustProvider configureTrustProvider(Tls tlsConfiguration) {
         return new NettyTrustProvider(trustProvider);
     }
 
+    private static void configureCipherSuites(SslContextBuilder sslContextBuilder, Tls tlsConfiguration) {
+        Optional.ofNullable(tlsConfiguration.cipherSuites())
+                .map(ciphers -> sslContextBuilder.ciphers(
+                        tlsConfiguration.cipherSuites().allowed(),
+                        new DenyCipherSuiteFilter(tlsConfiguration.cipherSuites().denied())));
+    }
+
+    private static void configureEnabledProtocols(SslContextBuilder sslContextBuilder, Tls tlsConfiguration) {
+
+        try {
+            var protocols = Optional.ofNullable(tlsConfiguration.protocols());
+
+            if (protocols.isPresent()) {
+                var allowedProtocols = Optional.ofNullable(protocols.get().allowed())
+                        .orElse(Arrays.stream(SSLContext.getDefault().getSupportedSSLParameters().getProtocols())
+                                .map(sslProtocol -> SslProtocol.getProtocolName(sslProtocol).get()).collect(Collectors.toSet()));
+                var deniedProtocols = Optional.ofNullable(protocols.get().denied())
+                        .orElse(Collections.EMPTY_SET);
+
+                var protocolsToUse = allowedProtocols.stream()
+                        .filter(Predicate.not(p -> deniedProtocols.contains(p)))
+                        .map(SslProtocol::getSslProtocol)
+                        .collect(Collectors.toList());
+
+                if (!protocolsToUse.isEmpty()) {
+                    sslContextBuilder.protocols(protocolsToUse);
+                }
+                else {
+                    LOGGER.warn("The protocols configuration you have in place has resulted in platform defaults being used");
```

I think we should fail here rather than just logging a warning.